### ISSUE TITLE
Accessibility issue: invalid `aria-colspan="0"` in `calcite-list-item-group`

### ISSUE THREAD

**Issue by brittaneyharkness** [Esri/calcite-components#14032]
**monday.com sync:** #11439757280

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Which Component

N/A, List Item Group

### Summary

While running Axe DevTools accessibility checks, an issue was flagged in the **Calcite `calcite-list-item-group` component**.

The component renders markup like this:

```html
<div role="cell" aria-colspan="0" class="heading">Bug Fixes</div>
```

### Actual Behavior

We are running Axe DevTools on all our GIS applications and we found that the ListGroup component we are using in our app is rendering an aria-colspan attribute where it equals "0". you can see in the screenshot: 

<img width="1901" height="939" alt="Image" src="https://github.com/user-attachments/assets/6af5095f-53b3-4570-8480-18e2278cc7ac" />

### Expected Behavior

I checked the ListGroup component for calcite 5.0 and I see its using MAX_COLUMNS for the ariaColSpan attribute and this is showing up as invalid when MAX_COLUMNS equals 0.

Would it be possible to add a conditional value like this? 
```
<div
  ariaColSpan={MAX_COLUMNS > 0 ? MAX_COLUMNS : undefined}
  class={CSS.heading}
  role="cell"
>
  {heading}
</div>
```

to prevent the component from rendering an invalid ARIA value

### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/components/list-item-group/

### Reproduction Steps

<CalciteListItemGroup heading={translateText("Bug Fixes")}>
                            <CalciteListItem 
                                label={translateText("Parcel Labeling")}
                                description={translateText("Fixed issues with parcel label classifications.")}>
                            </CalciteListItem>                           
</CalciteListItemGroup>

### Reproduction Version

3.0.3

### Priority impact

impact - p1 - need for current milestone

### Calcite package

- [x] @esri/calcite-components
- [x] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

Calcite (dev)

**Comment by geospatialem** [comment]
Spike needed to determine if `aria-colspan="0"` should be removed, or if the axe tool may not be giving out the correct information.
